src/annotator/highlight-clusters.tsx

```diff
@@ -213,6 +214,45 @@ export class HighlightClusterController implements Destroyable {
       );
     });
   }
+
+  /**
+   * Ensure that SVG <rect> elements are ordered correctly: inner (nested)
+   * highlights should be visible on top of outer highlights.
+   *
+   * All SVG <rect>s drawn for a PDF page are siblings. To ensure that the
+   * <rect>s associated with outer highlights don't show up on top of (and thus
+   * obscure) nested highlights, order the <rects> by their `data-nesting-level`
+   * value if they are not already.
+   */
+  _updateHighlightOrdering() {
+    const nestingLevel = (el: Element) =>
+      parseInt(el.getAttribute('data-nesting-level') ?? '0', 10);
+
+    // TODO: update `highlighter` to export some util functions to retrieve
```

Comment:
As this comment indicates, don't take any of the query selectors here as anything but PoC.